pkg/workload/workload.go

```diff
@@ -69,6 +69,13 @@ var (
 	}
 )
 
+// WorkloadReference is the full reference to Workload formed as <namespace>/< kueue.WorkloadName >.
+type WorkloadReference string
```

Comment:
```suggestion
type Reference string
```

This looks weird since this is called by `workload.WorkloadReference`.
@mimowo Do you have any expectations when you open an issue to request this cleanup?

Reply:
> @mimowo Do you have any expectations when you open an issue to request this cleanup?

I didn't think about this back then.

`WorkloadReference` pros:
- less changes when we move to another package (say API)
- more aligned with other analogous constants
cons:
-  for now "Workload" is redundant

I don't have any preference, I'm with workload.Reference for now.

Reply:
Ok, on the second thought I'm ok with WorkloadReference for consistency with other types. The redundancy in workload is not too bad imo.

From here I see two options:
1. expose the types into API packages
2. create dedicated impl packages per type and call the type, example "clusterqueue.Reference"

I'm actually leaning to (1.), because such type aliases are a common practice in k8s core, and so I think we could use this pattern in API too, examples:
- [ResourceName](https://github.com/kubernetes/kubernetes/blob/b65f712d2cb70a358e9c77d02b21874c14d3d1ba/pkg/apis/core/types.go#L5541-L5542)
- [FinalizerName](https://github.com/kubernetes/kubernetes/blob/b65f712d2cb70a358e9c77d02b21874c14d3d1ba/pkg/apis/core/types.go#L5607-L5608)